redpanda/templates/statefulset.yaml

```diff
@@ -109,6 +109,24 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: status.hostIP
+          startupProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                {{- if (include "admin-internal-tls-enabled" . |fromJson).bool }}
+                - >
+                  curl https://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview
+                  -svk --cacert /etc/tls/certs/{{ .Values.listeners.admin.tls.cert }}/ca.crt |
+                  grep '"is_healthy": true'
+                {{- else }}
+                - >
+                  curl -sv http://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview |
+                  grep '"is_healthy": true'
+                {{- end }}
```

Comment:
this snippet is now triplicated. it could be put into the `_helpers.tpl` and in each of these places it could be called as a template

Reply:
I think I'd like to leave this as-is, at this time, for readability. I'm also not convinced that one or more of these might change in the near future. Let's leave this for now and revisit after this chart gets more adoption and feedback. wdyt?